flowpm/raytracing.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Feb  3 12:29:19 2021
+
+@author: Denise Lanzieri
+"""
+import numpy as np
+import tensorflow as tf
+import tensorflow_addons as tfa
+import flowpm
+import flowpm.constants as constants
+
+
+def generate_matrix():
+  """ Generate the possible rotation matrices
+    """
+  x = np.asarray([1, 0, 0], dtype=int)
+  y = np.asarray([0, 1, 0], dtype=int)
+  z = np.asarray([0, 0, 1], dtype=int)
+  M_matrices = [np.asarray([x,y,z],dtype=int), np.asarray([x,z,y],dtype=int),np.asarray([z,y,x],dtype=int),np.asarray([z,x,y],dtype=int), \
+                     np.asarray([y,x,z],dtype=int), np.asarray([y,z,x],dtype=int)]
+
+  return M_matrices
+
+
+def shift():
+  """ Generate possible xy shifts
+    """
+  I = np.zeros(3)
+  facx = np.random.uniform(0, 1)
+  facy = np.random.uniform(0, 1)
+  xyshifts = [np.asarray([facx, facy, 0], dtype=float)]
+  return I + xyshifts
+
+
+def density_plane(state,
+                  nc,
+                  center,
+                  Matrix,
```

Comment:
Two things ^^'
1. we should rename the `Matrix` argument into `rotation` 
2. we should place the rotation and shift arguments at the end of the list, just before `name='...'`, and set them to `None` by default. The reason for moving them to the end, is because you don't want to separate thee `center` from the `width` arguments, together they define what the plane is going to look like

Reply:
done